### PR TITLE
Restore backward-compatibility for #6816

### LINKED ISSUE #6828: Backward compatibility for #6816
#6816 is not backward-compatible as the necessary changes in the tests and the [recently failing HDF5 test](https://cdash.kyomu.43-1.org/testDetails.php?test=17947376&build=2835) show. 
Examples are the `DataOutBase::write_*` functions.
We should keep the old signatures, deprecate them and forward to the new interface.

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #6828.

### CODE REVIEW

File: source/base/data_out_base.cc [hunk 1]
```diff
@@ -4398,6 +4576,42 @@ namespace DataOutBase
...
+      {
+        new_nonscalar_data_ranges[i] =
+          std::tuple<unsigned int,
+                     unsigned int,
+                     std::string,
+                     DataComponentInterpretation::DataComponentInterpretation>(
+            std::get<0>(nonscalar_data_ranges[i]),
+            std::get<1>(nonscalar_data_ranges[i]),
+            std::get<2>(nonscalar_data_ranges[i]),
+            DataComponentInterpretation::DataComponentInterpretation{});
```
Comment (davydden) [inline source/base/data_out_base.cc]: part_of_vector

File: source/base/data_out_base.cc [hunk 2]
```diff
@@ -4728,6 +4942,41 @@ namespace DataOutBase
...
```
Comment (davydden) [inline source/base/data_out_base.cc]: part_of_vector

File: source/base/data_out_base.cc [hunk 3]
```diff
@@ -5038,6 +5306,42 @@ namespace DataOutBase
...
```
Comment (davydden) [inline source/base/data_out_base.cc]: part_of_vector

File: source/base/data_out_base.cc [hunk 4]
```diff
@@ -5364,6 +5668,39 @@ namespace DataOutBase
...
```
Comment (davydden) [inline source/base/data_out_base.cc]: part_of_vector

File: source/base/data_out_base.cc [hunk 5]
```diff
@@ -6586,6 +6963,42 @@ namespace DataOutBase
...
```
Comment (davydden) [inline source/base/data_out_base.cc]: part_of_vector

File: source/base/data_out_base.cc [hunk 6]
```diff
@@ -7078,6 +7493,43 @@ DataOutBase::write_filtered_data(
...
+    {
+      new_nonscalar_data_ranges[i] =
+        std::tuple<unsigned int,
+                   unsigned int,
+                   std::string,
+                   DataComponentInterpretation::DataComponentInterpretation>(
+          std::get<0>(nonscalar_data_ranges[i]),
+          std::get<1>(nonscalar_data_ranges[i]),
+          std::get<2>(nonscalar_data_ranges[i]),
+          DataComponentInterpretation::DataComponentInterpretation{});
```
Comment (davydden) [inline source/base/data_out_base.cc]: part_of_vector

**Comment by davydden** [comment]
ps. thanks 👍

**Comment by masterleinad** [comment]
Fixed.

**Pull request merged**